src/admin/pages/Login.tsx

```diff
@@ -0,0 +1,104 @@
+import { ChangeEvent, FormEvent, useState } from "react";
+import { useLocalStorage } from "../../hooks/useLocalStorage";
+import { loginAdmin } from "../../network/adminApi";
+import { Jwt } from "../../network/jwt";
+import styles from "./Login.module.scss";
+
+interface LoginProps {
+  onLogin(jwt: Jwt): void;
+}
+
+export const Login = ({ onLogin }: LoginProps) => {
+  const [rememberUsername, setRememberUsername] = useLocalStorage<boolean>(
+    "admin.rememberUsername",
+    true
+  );
+  const [username, setUsername] = useLocalStorage<string>(
+    "admin.cacheUsername",
+    ""
+  );
+  const [password, setPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function changeUsername(e: ChangeEvent<HTMLInputElement>) {
+    const username = e.target.value;
+    setUsername(username, { remember: rememberUsername });
+    setErrorMessage("");
+  }
```

Comment:
Why update the stored username on every input? Wouldn't storing it on only a successful submit be enough?

Reply:
That's a good idea.